EXTENSION_FROM_SCRATCH.md

```diff
@@ -0,0 +1,254 @@
+
+But it doesn't _do_ anything yet.  Here is where you can start defining your own functionality.
+
+## Implement Other Interfaces and Extension Points
+
+If you want to handle REST Requests, implement the `ActionExtension` interface and override the `getExtensionRestHandlers()` method.  Pass a list of classes which will handle those requests.
+
+```java
+import org.opensearch.sdk.ActionExtension;
+
+public class CRUDExtension extends BaseExtension implements ActionExtension {
+
+    // keep constructor and main method from before and add the below
+
+    @Override
+    public List<ExtensionRestHandler> getExtensionRestHandlers() {
+        // we need to create this class next!
+        return List.of(new CrudAction());
+    }
+}
+```
+
+These classes must `implement ExtensionRestHandler`, which is a functional interface which requires the implementation of `public ExtensionRestResponse handleRequest(ExtensionRestRequest request)`.
+
+However the `BaseExtensionRestHandler` provides many useful methods for exception handling in requests including a `RouteHandler` class which eases logical separation of multiple `Route` choices.
+
+For the CRUD extension example, we'll implement one REST Route for each option and delegate to the appropriate handler function, although each one could be in its own file.
+
+```java
+import java.util.List;
+import java.util.function.Function;
+
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.extensions.rest.ExtensionRestResponse;
+import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.sdk.BaseExtensionRestHandler;
+import org.opensearch.sdk.RouteHandler;
+
+public class CrudAction extends BaseExtensionRestHandler {
+
+    @Override
+    protected List<RouteHandler> routeHandlers() {
+        return List.of(
+            new RouteHandler(Method.PUT, "/sample", createHandler),
+            new RouteHandler(Method.GET, "/sample/{id}", readHandler),
+            new RouteHandler(Method.POST, "/sample/{id}", updateHandler),
+            new RouteHandler(Method.DELETE, "/sample/{id}", deleteHandler)
+        );
+    }
+
+    Function<ExtensionRestRequest, ExtensionRestResponse> createHandler = (request) -> {
+        return new ExtensionRestResponse(request, RestStatus.OK, "To be implemented");
+    };
+
+    Function<ExtensionRestRequest, ExtensionRestResponse> readHandler = (request) -> {
+        return new ExtensionRestResponse(request, RestStatus.OK, "To be implemented");
+    };
+
+    Function<ExtensionRestRequest, ExtensionRestResponse> updateHandler = (request) -> {
+        return new ExtensionRestResponse(request, RestStatus.OK, "To be implemented");
+    };
+
+    Function<ExtensionRestRequest, ExtensionRestResponse> deleteHandler = (request) -> {
+        return new ExtensionRestResponse(request, RestStatus.OK, "To be implemented");
+    };
+}
+```
+
+## Use OpenSearch Clients to implement functionality
```

Comment:
Should we call it 
```suggestion
## Use SDK Clients to implement functionality
```

Reply:
How about "Java Clients for OpenSearch" since [that's their name](https://github.com/opensearch-project/opensearch-java)?  We may be removing these clients from SDK and we should document otherwise.